hera_cal/lstbin.py

```diff
@@ -579,110 +577,137 @@ def lst_bin_files(data_files, filetype='uvh5', input_cals=None, dlst=None, verbo
+    blgroups = [bls[i * (Nbls // Nblgroups + 1):(i + 1) * (Nbls // Nblgroups + 1)] for i in range(Nblgroups)]
+    blgroups = [blg for blg in blgroups if len(blg) > 0]
 
     # iterate over output LST files
     for i, f_lst in enumerate(file_lsts):
         utils.echo("LST file {} / {}: {}".format(i + 1, len(file_lsts), datetime.datetime.now()), type=1, verbose=verbose)
-
-        # create empty data_list and lst_list
-        data_list = []
-        file_list = []
-        flgs_list = []
-        lst_list = []
-
-        # locate all data files that fall within the range of lst for this output file
-        f_min = np.min(f_lst)
-        f_max = np.max(f_lst)
-        f_select = np.array(list(map(lambda d: list(map(lambda f: (f[1] > f_min) & (f[0] < f_max + atol), d)), data_times)))
-        if i == 0:
-            old_f_select = copy.copy(f_select)
-
-        # open necessary files, close ones that are no longer needed
-        # append necessary files to data_list
-        for j in range(len(data_files)):
-            nightly_data_list = []
-            nightly_flgs_list = []
-            nightly_lst_list = []
-            for k in range(len(data_files[j])):
-                if f_select[j][k] and data_status[j][k] is None:
-                    # open file
-                    utils.echo("Opening {}".format(data_files[j][k]), verbose=verbose)
-                    d, fl, ap, a, f, t, li, p = io.load_vis(data_files[j][k], return_meta=True, filetype=filetype)
-                    d.phase_type = 'drift'
-
-                    # open and apply calibration if provided
+        fmin = f_lst[0] - (dlst / 2 + atol)
+        fmax = f_lst[-1] + (dlst / 2 + atol)
+
+        # iterate over baseline groups (for memory efficiency)
+        data_conts, flag_conts, std_conts, num_conts = [], [], [], []
+        for bi, blgroup in enumerate(blgroups):
+            utils.echo("starting baseline-group {} / {}: {}".format(bi + 1, len(blgroups), datetime.datetime.now()), type=0, verbose=verbose)
+
+            # create empty data lists
+            data_list = []
+            file_list = []
+            flgs_list = []
+            lst_list = []
+     
+            # iterate over individual nights to bin
+            for j in range(len(data_files)):
+                nightly_data_list = []
+                nightly_flgs_list = []
+                nightly_lst_list = []
+
+                # iterate over files in each night, and open files that fall into this output file LST range
+                for k in range(len(data_files[j])):
+                    # unwrap la relative to itself
+                    larr = lst_arrs[j][k]
+                    tarr = time_arrs[j][k]
+                    larr[larr < larr[0]] += 2 * np.pi
+
+                    # check if this file has overlap with output file
+                    if larr[-1] < fmin or larr[0] > fmax:
+                        continue
+
+                    # if no overlap, get relevant time indicies
```

Comment:
I think you want to delete the phrase "if no overlap"... this only happens if there is overlap.